src/backend/components/py.tag/caliopen_tag/models_manager/manager.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""Caliopen user message qualification logic."""
+from __future__ import absolute_import, print_function, unicode_literals
+import logging
+import os
+from tempfile import NamedTemporaryFile
+from .data_manager import UsenetDataManager, ESDataManager
+from caliopen_storage.config import Configuration
+from ..utils import resources_path
+
+import fastText
+
+log = logging.getLogger(__name__)
+usenet_file = "usenet_data_{}.txt"
+
+
+class ModelManager(object):
+    """
+    Manage tagging model:
+     - training a new model
+    """
+
+    def __init__(
+            self,
+            cat="cat1",
+            use_pre=True,
+            epoch=15,
+            lr=0.1,
+            word_n_grams=2,
+            min_count=400,
+            dim=100,
+            loss="softmax",
+            thread=12,
+            neg=5):
+        self.cat = cat  # Check cat is available. Use int ?
+        self.use_pre = use_pre
+        self.epoch = epoch
+        self.lr = lr
+        self.wordNgrams = word_n_grams
+        self.minCount = min_count
+        self.dim = dim
+        self.loss = loss
+        self.thread = thread
+        self.neg = neg
+
+    def get_new_model(self, output):
+        """Get a new model and save it in resources/models/output.
+
+        It first fetches the data from cassandra and write it to a temporary
+        file, following the fastText format.
+        Then it pre-processes the data (tokenization + lowercase).
+        Then it trains a model using fastText and saves it at output location.
+        Finally it removes the temporary file.
+        """
+        self._write_training_data_to_file()
+        self._train_tagging_model(output)
+        self._remove_tempfile()
+
+    def _write_usenet_data(self, f):
```

Comment:
From my point of view these 2 methods (_write_usenet_data and _write_elastic_data) should not exists.

If to build the model we need 2 distincts data provider (usenet + elastic), we must combine them in a single DataProvider class.

My point is : give a single DataProvider instance to ModelManager.__init__ constructor, then the ModelManager will use this provider to iterate and format over items to build the model.

Then we can train different models with different DataProvider source

Reply:
I made some changes following this review. 

The DataManager takes a single DataProvider as constructor and simply iterates on it, as it is the general use case.

Additionally, the new MultipleSourceDataManager allows to make a wrapper of multiple providers and iterates on those successively.

Reply:
To learn a model combining data from Usenet and ES providers, we could do the following:
`
        usenet_manager = UsenetDataManager({})
        usenet_manager.prepare("/path/to/data.txt")

        config = Configuration('global').configuration
        es_manager = ESDataManager(config)
        query = es_manager.get_query()
        es_manager.prepare(query, index=None, doc_type='indexed_message')

        provider = MultipleSourceDataManager([usenet_manager, es_manager])

        for item in provider.next():
                print(item)
`